### PR TITLE
Add initial html -> md conversion and axios http client retry

### PR DESCRIPTION
**Pull request by ccho-mongodb**
https://github.com/mongodben/mongodb-oracle/issues/10

### CODE REVIEW

File: app/generate-index/gen-site-data.ts [hunk 1]
```diff
@@ -56,19 +58,28 @@ export async function getHtmlPages(urlList: string[]) {
 // TODO: make this script more fault tolerant w retry behavior.
 // right now fails when there are momentary internet drops
 export async function getPageData(url: string) {
+  axiosRetry(axios, { retryDelay: axiosRetry.exponentialDelay });
```
Comment (mongodben) [inline app/generate-index/gen-site-data.ts:61]: nice find!

File: app/generate-index/gen-site-data.ts [hunk 2]
```diff
@@ -56,19 +58,28 @@ export async function getHtmlPages(urlList: string[]) {
...
 
-export function snootyHtmlToText(html: string) {
+function getPageBody(html: string) {
   const dom = new JSDOM(html);
   const $ = require("jquery")(dom.window);
 
-  $("html").find("header").remove();
-  $("html").find("footer").remove();
-  $("html").find("nav").remove();
+  const content = $("main").html();
```
Comment (mongodben) [inline app/generate-index/gen-site-data.ts:70]: looking at the MD output, all the headers are postfixed with the string `[](#introduction \"Permalink to this heading\")` 

could you remove this in the parsing?

**Review by mongodben — CHANGES_REQUESTED** [review]
back to you w 1 request

File: app/generate-index/gen-site-data.ts (hunk 2 above)
Comment (ccho-mongodb) [reply app/generate-index/gen-site-data.ts:70]: There are plenty of issues with the parsing that aren't trivial to solve in this PR, working on them in another branch.

**Comment by mongodben** [comment]
superceded by https://github.com/mongodben/mongodb-oracle/pull/42

**Pull request closed without merging**